### ISSUE TITLE
Inaccuracy in stored floating point numbers. 

### ISSUE THREAD

**Issue by ReinisSprogis** [objectbox/objectbox-dart#641]
### Build info

- objectbox version: 4.0.1
- Flutter/Dart version: 3.22.1
- Build OS:  Windows 10
- Deployment OS or device: Windows 10

### Steps to reproduce

1. Save different double values in database.
2. See incorrect value saved.

### Expected behavior

Saving values that are provided to save.
Saving | Saved
16.9 -> 16.9
16.4 ->16.4
17.4 -> 17.4
19.5 -> 19.5
17.1 -> 17.1

### Actual behavior

Almost all floating point values are incorrectly saved with many decimal places
Saving | Saved
16.9 -> 16.899999618530273
16.4 -> 16.399999618530273
17.4 -> 17.399999618530273
19.5 -> 19.5
17.1 -> 17.100000381469727

### Code

```dart

@Entity()
class DataValue {
  @Id()
  int id = 0;

  @Property(type: PropertyType.float)
  double? value;

  @Property(type: PropertyType.date)
  DateTime? date;

  DataValue({this.value, this.date,});

  final station = ToOne<Station>();
  }
```

**Comment by ReinisSprogis** [comment]
Is this expected behavior? Bug? Or I am doing something wrong?

**Comment by greenrobot** [comment]
Try removing `@Property(type: PropertyType.float)` so it uses the default type for double, which is also double (64 bit) and not float (32 bit).

**Comment by ReinisSprogis** [comment]
Thanks for reply. When I remove ```@Property(type: PropertyType.float)```  I get error when try to update values. ```[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: ObjectBoxException: failed to create store: Existing Property value (2, Float) of entity DataValue is not compatible with the new type Double```

```dart
    for(var station in localStations){
      for(var observation in station.observations){
        if(observation.data.isNotEmpty){
          continue;
        }

        List<DataValue> updatedDataList = [];
        for(var data in observation.data){

          if(data.value == null){
            continue;
          }
          data.value = roundToPrecision(data.value!, 1);
          updatedDataList.add(data);
        }
       objectbox.store.box<DataValue>().putMany(updatedDataList);
      }
    }
```

I have 180 million data points, I have to loop through them all and round them. Hoping they will round correctly. Any suggestions how to resolve that error? I was looking through https://docs.objectbox.io/advanced/data-model-updates But couldn't find answer.

**Comment by ReinisSprogis** [comment]
I think I could create another database and rewrite everything to it with rounded values. Then just delete old one.

**Comment by greenrobot** [comment]
Yes, better reimport all the data - the accuracy is already lost...

**Comment by ReinisSprogis** [comment]
This is scarped data from public db, I saved straight into ObjectBox. I had a feeling I have to save in json in parallel just in case, but I didn't. So now I either have to scrape again or just trust rounded values. I ended up creating new model ```Value``` and re-save to that in the same db. I will stick with rounded values for now. However when I run ```objectbox.store.box<DataValue>().removeAll()``` on old data database still is double the size. But ```objectbox.store.box<DataValue>().getAll()``` shows 0 values. I understand that it keeps memory allocated? So now to get back to 25GB database instead of 55 I need to copy all data over to new database?  FAQ https://docs.objectbox.io/faq#how-does-objectbox-use-disk-space-and-can-i-reclaim-disk-space Some ```reclaimMemory()``` method would be appreciated so wouldn't be need to do that.
